cpp/mrc/src/internal/grpc/client_streaming.hpp

```diff
@@ -195,10 +197,10 @@ class ClientStream : private Service, public std::enable_shared_from_this<Client
         while (s.is_subscribed())
         {
             CHECK(m_stream);
-            Promise<bool> read;
+            auto* wrapper = new PromiseWrapper("Client::Read");
```

Comment:
Question: Do we still need to use `new` here even with `PromiseWrapper` and the changes to `WritesDone`?

Reply:
Unfortunately yes, that is the difference between this PR and #389 
With the `PromiseWrapper` and `WritesDone` changes I was no longer able to reproduce the segfault with clang builds. However I was still getting the segfault with the gcc code-coverage builds, and it was in the same place as when we got it in the clang builds:
```
2023-09-19T13:52:28.4885028Z *** Aborted at 1695131547 (unix time) try "date -d @1695131547" if you are using GNU date ***
2023-09-19T13:52:28.4885919Z PC: @                0x0 (unknown)
2023-09-19T13:52:28.4886833Z *** SIGSEGV (@0x0) received by PID 27049 (TID 0x7f5f99ffb000) from PID 0; stack trace: ***
2023-09-19T13:52:28.4915867Z     @     0x7f60cb48d197 google::(anonymous namespace)::FailureSignalHandler()
2023-09-19T13:52:28.4916811Z     @     0x7f60cb822420 (unknown)
2023-09-19T13:52:28.4917647Z     @     0x7f60cb43ff2e boost::fibers::wait_queue::notify_all()
2023-09-19T13:52:28.4918601Z     @     0x7f60cb43dee3 boost::fibers::condition_variable_any::notify_all()
2023-09-19T13:52:28.4919590Z     @     0x7f60cc3f2e9a boost::fibers::condition_variable::notify_all()
2023-09-19T13:52:28.4920691Z     @     0x7f60cc3f30a9 boost::fibers::detail::shared_state_base::mark_ready_and_notify_()
2023-09-19T13:52:28.4921716Z     @     0x7f60cc6daa2b boost::fibers::detail::shared_state<>::set_value_()
2023-09-19T13:52:28.4925964Z     @     0x7f60cc6da200 boost::fibers::detail::shared_state<>::set_value()
2023-09-19T13:52:28.4926906Z     @     0x7f60cc6d9999 boost::fibers::promise<>::set_value()
2023-09-19T13:52:28.4927846Z     @     0x7f60cc6d8ca7 mrc::rpc::PromiseWrapper::set_value()
2023-09-19T13:52:28.4929901Z     @     0x7f60cc3fd93a mrc::rpc::PromiseHandler::on_data()
```

Reply:
got it